internal/tracker/tracker.go

```diff
@@ -314,20 +302,23 @@ func filterRecords(records []taskRecord, prune bool) []taskRecord {
 	return relevant
 }
 
-// setExpiration sets the expires_on attribute on records. The expires_on value for record N is the
-// effective_on value of the n-1 record. The first record on the list does not contain an expires_on
-// value since there is no newer record that invalidates it in the future.
-// TODO: Probably need to compute the expires_on value without requiring the "effective_on" value so
-// we don't have to always require both values. But this may be required during some transition
-// period.
-func (t *Tracker) setExpiration() {
+// setExpiration sets the expires_on attribute on records using a duration-based approach.
+// Each record expires after a configurable duration based on inEffectDays from when it's added.
+// The most recent record (index 0) gets no expiration date, making it the current active record.
+func (t *Tracker) setExpiration(inEffectDays int) {
+	expirationDuration := time.Duration(inEffectDays) * oneDay // Use --in-effect-days flag value
+	now := time.Now().UTC().Round(oneDay)
```

Comment:
Maybe this is nicer:

```suggestion
	expiresOn := time.Now().UTC().Add(expirationDuration)
```

then use it on line 319.

Ps, what is the idea behine the `.Round(oneDay)`? What does that do actually?